src/table/virtualized-table/hooks/use-measure-text.ts

```diff
@@ -2,14 +2,61 @@ import { useMemo } from 'react';
 import { memoize } from 'qlik-chart-modules';
 import { MAX_NBR_LINES_OF_TEXT } from '../constants';
 
+type EstimateLineCountProps = {
+  text: string;
+  maxWidth: number;
+  isNumeric?: boolean;
+};
+
+export type EstimateLineCount = ({ text, maxWidth, isNumeric }: EstimateLineCountProps) => number;
+
 export interface MeasureTextHook {
   estimateWidth: (length: number) => number;
   measureText: (text: string) => number;
-  estimateLineCount: (text: string, maxWidth: number, isNumeric?: boolean) => number;
+  estimateLineCount: EstimateLineCount;
 }
 
 const MAGIC_DEFAULT_CHAR = 'N';
 
+function getNextLine(text: string, maxWidth: number, fixedMeasureText: (text: string) => number) {
+  let left = 0;
+  let right = text.length;
+
+  if (text.length <= 1) {
+    return null;
+  }
+
+  if (fixedMeasureText(text) <= maxWidth) {
+    return null;
+  }
+
+  while (left <= right) {
+    const m = Math.floor((left + right) / 2);
+    const chunk = text.slice(0, m);
+    const prevChunk = text.slice(0, m - 1);
+    const width = fixedMeasureText(chunk);
+    const prevWidth = fixedMeasureText(prevChunk);
+
+    if (width > maxWidth && prevWidth <= maxWidth) {
+      return text.slice(m - 1) || null;
+    }
+
+    if (width === maxWidth) {
+      return text.slice(m) || null;
+    }
+
+    if (width < maxWidth) {
+      // Search right
+      left = m + 1;
+    } else {
+      // Search left
+      right = m;
+    }
+  }
+
+  return null;
```

Comment:
seems like this row is not covered in the test, is it testable?

Reply:
Good find! But no, my thinking is that it's not. The only time it would reach here is when `fixedMeasureText(text) <= maxWidth` is true. But there is a guard for that before the while statement as a performance improvement.

I will however do some more digging into before my final conclusion.